### PR TITLE
Slaughter Demon

### PR DESCRIPTION
**Pull request by ghost**
New side event/antagonist. Simple animal that can use pools of blood as
portals to move around the station. Spooky scary.

-CtrlClick on an adjacent blood pool (or turf with blood on it) to
"enter" (start an infinite duration jaunt
-CtrlClick on a blood pool while jaunted to exit at that pool
-If you're pulling a mob while you enter a pool, the mob gets pulled in
with you, and you devour them, healing yourself
-Night vision

Adds an event (admin only) that spawns one.

Adds wizard item for summoning it. Has objective to kill the wizard (and everyone else)

If someone could help me with mind stuff so I can track how many people it eats that'd be great.

Webm

http://www.singulo.io/tg/src/1433549232359.webm

### CODE REVIEW

**Comment by Cheridan** [comment]
*teleports behind u*
*unsteahtes claws*

nothin personnel, kid

**Comment by KorPhaeron** [comment]
Do I get a prize for edgiest commit of the year?

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+				src.adjustBruteLoss(-1000)
+				kidnapped.ghostize()
+				qdel(kidnapped)
+				src.devoured++
+				src.kidnapped = null
+				src.eating = FALSE
+			src.notransform = 0
+
+/mob/living/simple_animal/slaughter/proc/phasein(var/obj/effect/decal/cleanable/B)
+	var/mob/living/simple_animal/slaughter/S = src
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: what are you doing.
why are you typecasting src.

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+		src.client.eye = src
+		var/list/voice = list('sound/hallucinations/behind_you1.ogg','sound/hallucinations/im_here1.ogg','sound/hallucinations/turn_around1.ogg','sound/hallucinations/i_see_you1.ogg')
+		playsound(get_turf(src), pick(voice),50, 1, -1)
+		src.visible_message("<span class='warning'><B>The [src] rises out of the pool of blood!</B>")
+		playsound(get_turf(src), 'sound/effects/slosh.ogg', 50, 1, -1)
+		qdel(S.holder)
+
+/obj/effect/decal/cleanable/blood/CtrlClick(var/mob/user)
+	..()
+	var/obj/effect/decal/cleanable/B = src
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: >typecasting src

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+		var/mob/living/simple_animal/slaughter/S = user
+		if(S.phased)
+			S.phasein(B)
+		else
+			S.phaseout(B)
+
+
+/obj/effect/decal/cleanable/trail_holder/CtrlClick(var/mob/user)
+	..()
+	var/obj/effect/decal/cleanable/B = src
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: >typecasting src

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+		if(S.phased)
+			S.phasein(B)
+		else
+			S.phaseout(B)
+
+
+
+/turf/CtrlClick(var/mob/user)
+	..()
+	var/turf/T = src
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: >typecasting src

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: Good question

**Comment by Incomptinence** [comment]
John Stalvern waited. The lights above him blinked and sparked out of the air. There were demons in the base. He didn’t see them, but had expected them now for years. His warnings to Cernel Joson were not listenend to and now it was too late. Far too late for now, anyway.
    John was a space marine for fourteen years. When he was young he watched the spaceships and he said to dad “I want to be on the ships daddy.”
    Dad said “No! You will BE KILL BY DEMONS”
    There was a time when he believed him. Then as he got oldered he stopped. But now in the space station base of the UAC he knew there were demons.
    “This is Joson” the radio crackered. “You must fight the demons!”
    So John gotted his palsma rifle and blew up the wall.
    “HE GOING TO KILL US” said the demons
    “I will shoot at him” said the cyberdemon and he fired the rocket missiles. John plasmaed at him and tried to blew him up. But then the ceiling fell and they were trapped and not able to kill.
    “No! I must kill the demons” he shouted
    The radio said “No, John. You are the demons”
    And then John was a zombie.

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+	var/playstyle_string = "<B>You are the Slaughter Demon, a terible creature from another existence. You have a single desire: To kill.  \
+						You may Ctrl+Click on blood pools to travel through them, appearing and dissaapearing from the station at will. \
+						Pulling a dead or critical mob while you enter a pool will pull them in with you, allowing you to feast. </B>"
+
+
+/mob/living/simple_animal/slaughter/death()
+	..(1)
+	new /obj/effect/decal/cleanable/blood (src.loc)
+	new /obj/item/weapon/demonheart (src.loc)
+	visible_message("<span class='danger'>The [src] screams in anger as it's form collapes into a pool of viscera.</span>")
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: its

File: code/modules/mob/living/simple_animal/slaughter/slaughter.dm [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+////////////////////The Powers
+
+/mob/living/simple_animal/slaughter/proc/phaseout(var/obj/effect/decal/cleanable/B)
+	var/turf/mobloc = get_turf(src.loc)
+	var/turf/bloodloc = get_turf(B.loc)
+	if(Adjacent(bloodloc))
+		src.notransform = TRUE
+		spawn(0)
+			src.visible_message("The [src] sinks into the pool of blood, and out of existence.")
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/slaughter/slaughter.dm]: "and out of existence" sounds kind of awkward here it could probably do without that

**Comment by Jordie0608** [comment]
What happens if all blood spots are cleaned up once the demon has started jaunting? Will it just have to wait around until someone gets hurt?

**Comment by KorPhaeron** [comment]
@Jordie0608 

Yes. The janitor has won at that point (though someone will always get hurt)

**Comment by Iamgoofball** [comment]
needs ERT Janitor role

**Comment by Stickymayhem** [comment]
Make it alt+click to use blood portals so it's consistent with ventcrawling.

Pretty cool though we need more side antags.

**Comment by KorPhaeron** [comment]
@Stickymayhem I had it as altclick initially but you'll already be hitting ctrl to pull mobs, so your finger will already be there/less combos to use.

**Comment by xxalpha** [comment]
Yes.

**Comment by Stickymayhem** [comment]
What if someone is on some blood? Makes it a lot more likely to accidentally use the portal under them than drag them into it.

**Comment by WJohn** [comment]
I think entering a blood pool should be alt clicking instead of ctrl clicking. For consistency reasons, because that's how entering vents is and ctrl click is already heavily used for dragging.

**Comment by KorPhaeron** [comment]
@Stickymayhem 

Good question. I hadn't considered that/run into the problem when running around trying to click on things.

@WJohn 

See my reply to sticky about having one button combo. Maybe I'm the only one who finds it awkward to use the alt key, I much prefer resting my thumb on the spacebar

**Comment by Niknakflak** [comment]
Janitors can finally be the hero the station needs, but not the one they deserve

**Comment by KorPhaeron** [comment]
Until a janiborg replaces the janitor and steamrolls the antag just like borgs do every round.

**Comment by AnturK** [comment]
It probably should start in blood-dimension. Spooks breaking windows to get on station makes them less scary.

**Comment by KorPhaeron** [comment]
@Anturk

Ir does

**Comment by Xhuis** [comment]
What why? This was a good mini-antag.

**Comment by Iamgoofball** [comment]
>closed
>deleted

why

**Comment by KorPhaeron** [comment]
@Iamgoofball 

Remade it because I couldn't resolve the merge conflicts properly

**Comment by PKPenguin321** [comment]
@Iamgoofball 
https://github.com/tgstation/-tg-station/pull/9839

**Pull request closed without merging**